app/models/author.rb

```diff
@@ -170,14 +170,17 @@ def assign_pub(pub)
         return
     end
     raise 'Author must be saved before association' unless persisted?
-    pub.contributions.find_or_create_by!(author_id: id) do |contrib|
+
+    contribution = pub.contributions.find_or_initialize_by(author_id: id) do |contrib|
       contrib.assign_attributes(
         cap_profile_id: cap_profile_id,
         featured: false, status: 'new', visibility: 'private'
       )
     end
-    pub.pubhash_needs_update! # Add to pub_hash[:authorship]
-    pub.save! # contrib.save! not needed
+    return true unless contribution.new_record?
```

Comment:
This method is documented as returning a `Contribution`, but here you're returning a `Boolean`.

Reply:
ah yes, missed that -- turns out it wasn't actually returning a contribution before either (and the code doesn't care so it's never using the return value anyway), but fixed it to match the signature